### PR TITLE
Switched from miniconda-setup to micromamba-setup for mypy workflow

### PR DESCRIPTION
**Pull request by HangenYuu**
## Description

Switch from using setup-miniconda to setup-micromamba in GitHub Actions for 3 current workflows

## Related Issue

- [ ] Closes #
- [ ] Related to #775 

## Checklist

- [x] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [x] Included tests that prove the fix is effective or that the new feature works
- [ ] Added necessary documentation (docstrings and/or example notebooks)
- [ ] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Type of change

- [x] New feature / enhancement
- [ ] Bug fix
- [ ] Documentation
- [ ] Maintenance
- [ ] Other (please specify):

### CODE REVIEW

**Comment by HangenYuu** [comment]
mypy.yml works as expected. Moving on to other workflows 👍.

**Comment by HangenYuu** [comment]
This is this PR current output of the tests:
image

image
I wonder whether I should set this config flag directly in `test.yml`?

**Comment by ricardoV94** [comment]
CC @maresb @lucianopaz

**Comment by maresb** [comment]
@HangenYuu, any reason you closed this? I went on vacation right after you created this. I was hoping that @lucianopaz would be able to look at the BLAS flags error.

**Comment by HangenYuu** [comment]
I honestly did not know why too. I even deleted the branch. Let's me reopen it here.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -138,7 +140,7 @@ jobs:
           echo "id=$MATRIX_ID" >> $GITHUB_OUTPUT
 
       - name: Install dependencies
-        shell: bash -l {0}
+        shell: micromamba-shell {0}
         run: |
           mamba install --yes -q "python~=${PYTHON_VERSION}=*_cpython" mkl numpy scipy pip mkl-service graphviz cython pytest coverage pytest-cov pytest-benchmark pytest-mock sympy
```
Comment (lucianopaz) [inline .github/workflows/test.yml:145]:
```suggestion
          micromamba install --yes -q "python~=${PYTHON_VERSION}=*_cpython" mkl numpy scipy pip mkl-service graphviz cython pytest coverage pytest-cov pytest-benchmark pytest-mock sympy
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -138,7 +140,7 @@ jobs:
...
           if [[ $INSTALL_NUMBA == "1" ]]; then mamba install --yes -q -c conda-forge "python~=${PYTHON_VERSION}=*_cpython" "numba>=0.57"; fi
```
Comment (lucianopaz) [inline .github/workflows/test.yml:146]:
```suggestion
          if [[ $INSTALL_NUMBA == "1" ]]; then micromamba install --yes -q -c conda-forge "python~=${PYTHON_VERSION}=*_cpython" "numba>=0.57"; fi
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -175,55 +177,57 @@ jobs:
...
+        uses: mamba-org/setup-micromamba@v1
+        with:
+          environment-name: pytensor-test
+          micromamba-version: "latest"
+          init-shell: bash
+          post-cleanup: "all"
+      - name: Install dependencies
+        shell: micromamba-shell {0}
+        run: |
+          mamba install --yes -q -c conda-forge "python~=${PYTHON_VERSION}=*_cpython" mkl numpy scipy pip mkl-service cython pytest "numba>=0.57" jax jaxlib pytest-benchmark
```
Comment (lucianopaz) [inline .github/workflows/test.yml:202]:
```suggestion
          micromamba install --yes -q -c conda-forge "python~=${PYTHON_VERSION}=*_cpython" mkl numpy scipy pip mkl-service cython pytest "numba>=0.57" jax jaxlib pytest-benchmark
```

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -175,55 +177,57 @@ jobs:
...
+          pip install -e ./
+          mamba list && pip freeze
```
Comment (lucianopaz) [inline .github/workflows/test.yml:204]:
```suggestion
          micromamba list && pip freeze
```

**Review by lucianopaz — CHANGES_REQUESTED** [review]
@HangenYuu, it looks like your installation [isn't setting up the mamba command](https://github.com/pymc-devs/pytensor/actions/runs/9440568604/job/26000091232?pr=781#step:5:93). Instead of erroring out there, the jobs seems to continue and pip install pytensor. Without mkl, the jobs terminates because it does not find blas ldflags.

You should change all calls to `mamba` with `micromamba`. I suggested the few I could here, but there are more in the yml job files.

**Comment by HangenYuu** [comment]
@lucianopaz I made the changes and the tests seem to be passed already. Honestly did not expect `micromamba` to be used instead of `mamba`, but that was just my assumption from using `miniconda`, which still uses `conda` to install package.

**Review by twiecki — APPROVED** [review]

**Review by lucianopaz — APPROVED** [review]
LGTM

**Pull request merged**